core/store/src/adapter/trie_store.rs

```diff
@@ -188,30 +198,23 @@ impl<'a> TrieStoreUpdateAdapter<'a> {
 /// It is kept out of `TrieStoreAdapter`, so that `TrieStoreUpdateAdapter` can use it without
 /// cloning `store` each time, see https://github.com/near/nearcore/pull/12232#discussion_r1804810508.
 pub fn get_shard_uid_mapping(store: &Store, child_shard_uid: ShardUId) -> ShardUId {
+    maybe_get_shard_uid_mapping(store, child_shard_uid).unwrap_or(child_shard_uid)
+}
+
+/// Get the `ShardUId` mapping for child_shard_uid. If the mapping does not exist, return None.
+fn maybe_get_shard_uid_mapping(store: &Store, child_shard_uid: ShardUId) -> Option<ShardUId> {
     store
         .get_ser::<ShardUId>(DBCol::StateShardUIdMapping, &child_shard_uid.to_bytes())
         .unwrap_or_else(|_| {
             panic!("get_shard_uid_mapping() failed for child_shard_uid = {}", child_shard_uid)
         })
```

Comment:
nit: I think it's not possible to have `child_shard_uid` be equal to mapped shard_uid, but we could add a debug assert

Reply:
Alright! Let me include this as part of a follow up PR. Will tag this comment there